validation/src/main/java/io/serverlessworkflow/validation/state/StatesValidator.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022-Present The Serverless Workflow Specification Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.serverlessworkflow.validation.state;
+
+import io.serverlessworkflow.api.Workflow;
+import io.serverlessworkflow.api.interfaces.State;
+import io.serverlessworkflow.api.states.DefaultState.Type;
+import io.serverlessworkflow.api.validation.ValidationError;
+import java.util.Collection;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+
+public final class StatesValidator {
+  private final Map<Type, StateValidator> stateValidators = new EnumMap<>(Type.class);
+
+  private final Collection<ValidationError> validationErrors;
+
+  public StatesValidator(Collection<ValidationError> validationErrors) {
+    this.validationErrors = validationErrors;
+
+    stateValidators.put(Type.EVENT, new EventStateValidator(validationErrors));
+    stateValidators.put(Type.OPERATION, new OperationStateValidator(validationErrors));
+    stateValidators.put(Type.SWITCH, new SwitchStateValidator(validationErrors));
+    stateValidators.put(Type.SLEEP, new SleepStateValidator(validationErrors));
+    stateValidators.put(Type.PARALLEL, new ParallelStateValidator(validationErrors));
+    stateValidators.put(Type.INJECT, new InjectStateValidator(validationErrors));
+    stateValidators.put(Type.FOREACH, new ForEachStateValidator(validationErrors));
+    stateValidators.put(Type.CALLBACK, new CallbackStateValidator(validationErrors));
+  }
+
+  public boolean hasValidStates(Workflow workflow) {
+    boolean finalValidationResult = true;
+
+    List<State> states = workflow.getStates();
+    if (states != null && !states.isEmpty()) {
+      for (State state : states) {
+        StateValidator stateValidator = stateValidators.get(state.getType());
+
+        if (stateValidator != null) {
+          finalValidationResult =
+              finalValidationResult && stateValidator.isValidState(workflow, state);
```

Comment:
so, at the moment a validator fail, we cancel the whole validation process?
I guess this is intentional, but just double checking. 
Cancelling the validation process when the first state that has a validation error is found will make the validationErrors list not complete.

Reply:
No. The idea is to return false, which means "doesn't have valid states", but continue with the validation.
Currently the return value is being ignored, but that's just because it's not fully implemented yet.

Reply:
Ok, then, you need to change this code because stateValidator.isValidState(workflow, state) is not going to be invoked if previous state returned false. 
you should write` finalValidationResult&=stateValidator.isValidState(workflow, state);`